docs/source/walkthrough.rst

```diff
@@ -1,3 +1,781 @@
+..  3. annoying but not useful warning from git-annex 
+.. TODO after copying:
+..  1. check the tracked changes!
+
+.. dropdown:: Warning regarding TemplateFlow? Fine to toy BIDS App!
+
+    You may receive this warning from ``babs-init`` if you did not set up environment variable ``$TEMPLATEFLOW_HOME``::
+
+        UserWarning: Usually BIDS App depends on TemplateFlow, but environment variable `TEMPLATEFLOW_HOME` was not set up.
+        Therefore, BABS will not bind its directory or inject this environment variable into the container when running the container. This may cause errors.
+
+    This is totally fine to toy BIDS App, and it won't use TemplateFlow.
+    However, a lot of BIDS Apps would use it. Make sure you set it up when you use those BIDS Apps.
+
+
+It's very important to check if the generated ``singularity run`` command is what you desire:
+
+..  code-block:: console
+
+    singularity run --cleanenv \
+        -B ${PWD} \
+        containers/.datalad/environments/toybidsapp-0-0-7/image \
+        inputs/data/BIDS \
+        outputs \
+        participant \
+        --no-zipped \
+        --dummy 2 \
+        -v \
+        --participant-label "${subid}"
+
+
+As you can see, BABS has automatically handled the positional arguments of BIDS App (i.e., input directory,
+output directory, and analysis level - 'participant'). ``--participant-label`` is also covered by BABS, too.
+
+.. developer's note: below is generated based on `tree -L 3 .`
+
+.. dropdown:: What's inside the created BABS project ``my_BABS_project``?
+
+    ..  code-block:: console
+
+        .
+        ├── analysis
+        │   ├── CHANGELOG.md
+        │   ├── code
+        │   │   ├── babs_proj_config.yaml
+        │   │   ├── babs_proj_config.yaml.lock
+        │   │   ├── check_setup
+        │   │   ├── participant_job.sh
+        │   │   ├── README.md
+        │   │   ├── submit_job_template.yaml
+        │   │   ├── sub_ses_final_inclu.csv
+        │   │   └── toybidsapp-0-0-7_zip.sh
+        │   ├── containers
+        │   ├── inputs
+        │   │   └── data
+        │   ├── logs
+        │   └── README.md
+        ├── input_ria
+        └── output_ria
+    
+    Here, ``analysis`` is a DataLad dataset that includes generated scripts in ``code/``,
+    cloned container DataLad dataset ``containers/``, and cloned input dataset in ``inputs/data``.
+    Input and output RIA stores (``input_ria`` and ``output_ria``) are DataLad siblings of ``analysis``.
+    When job running, inputs are cloned from input RIA store,
+    and results and provenance will be pushed to output RIA store. 
+
+
+Step 2.2. Use ``babs-check-setup`` to make sure it's good to go
+--------------------------------------------------------------------
+It's important to let BABS checks if everything has been correctly set up. In addition,
+it's a good idea to run a toy, test job to make sure the environment you specified in the YAML file
+is working as expected.
```

Comment:
Hmmm this might not be accurate, e.g., `babs-check-setup` test job won't check if `singularity_run` or `zip_foldernames` are workable. I'll mostly keep my original words here.